nanoFramework.Json/JsonConvert.cs

```diff
@@ -1088,32 +1088,38 @@ private static LexToken GetNextTokenInternal(ref int jsonPos, ref byte[] jsonByt
                         return EndToken(sb);
                     }
 
-                    //TODO: replace with a mapping array? This switch is really incomplete.
-                    switch (ch)
+                    if(ch == 'u')
                     {
-                        case 't':
-                            ch = '\t';
-                            break;
-
-                        case 'r':
-                            ch = '\r';
-                            break;
-
-                        case 'n':
-                            ch = '\n';
-                            break;
-
-                        case 'u':
-                            unicodeEncoded = true;
-                            break;
-
-                        case '"':
-                            ch = '"';
-                            break;
-
-                        default:
-                            throw new DeserializationException();
+                        unicodeEncoded = true;
                     }
+
```

Comment:
Why all these are being commented out? If they aren't needed better delete them...

Reply:
To be honest, bad habit I picked up from school (leaving previous code since we did not use change tracking), i will fix that and commit.

Reply:
also while going over it again, i found yet another situation where it is not de-serializing correctly, going to add more tests to more thoroughly cover the characters instead of just checking the " and \ characters in the tests.

Reply:
I have committed the updated/added tests and the modified logic in StringConverter and JsonConvert.  There are still two tests not correctly passing (one of which I added).  I am not sure if I am incorrect about the test values, as the same problem seems to exist in the test SerializeCosmosDbObject_02 which is failing for removed quotes as well, though the test SerializeCosmosDbObject_01 which uses almost the same string, seems to work fine.  I think another set of eyes may be necessary.

Reply:
alright final code has been committed that replaces the quotes that should not be removed during de-serialization of strings that are part of complex objects which also contain an escaped quote at the start and end of the string. all tests should now be passing.